src/Microsoft.AspNet.Mvc.ViewFeatures/Internal/ValidationHelpers.cs

```diff
@@ -3,27 +3,45 @@
 
 using System;
 using System.Collections.Generic;
+using System.Diagnostics;
 using System.Linq;
 using Microsoft.AspNet.Mvc.ModelBinding;
 
 namespace Microsoft.AspNet.Mvc.ViewFeatures.Internal
 {
-    internal static class ValidationHelpers
+    public static class ValidationHelpers
     {
-        public static string GetUserErrorMessageOrDefault(ModelError modelError, ModelStateEntry entry)
+        public static string GetModelErrorMessageOrDefault(ModelError modelError)
         {
+            Debug.Assert(modelError != null);
+
             if (!string.IsNullOrEmpty(modelError.ErrorMessage))
             {
                 return modelError.ErrorMessage;
             }
 
-            if (entry == null)
+            // Default in the ValidationSummary case is no error message.
+            return string.Empty;
+        }
+
+        public static string GetModelErrorMessageOrDefault(
+            ModelError modelError,
+            ModelStateEntry containingEntry,
+            ModelExplorer modelExplorer)
+        {
+            Debug.Assert(modelError != null);
+            Debug.Assert(containingEntry != null);
+            Debug.Assert(modelExplorer != null);
+            Debug.Assert(modelExplorer.Metadata?.ModelBindingMessageProvider?.ValueIsInvalidAccessor != null);
```

Comment:
nit: odd to put asserts about an unrelated class here. We're not trying to make the code provably correct :+1: